tests/ui/useless_conversion.rs

```diff
@@ -194,11 +194,24 @@ mod issue11300 {
         assert_eq!(i.len(), 3);
     }
 
+    trait Helper {}
```

Comment:
Can you also add a test for something like `trait Helper<T: ?Sized>`? Basically any trait with generics.

A quick look at the new version makes me think this *should* work, but a test would be great

Reply:
I changed the tests a bit, is this what you had in mind? Or something else?

Reply:
I was thinking a test that would lint, basically seeing if it would substitute the `T` in `Helper<T>` properly

Reply:
I see. Well, I think the tests I just added are also a good addition since that found a bug while I was testing stuff so I'm just going to leave them and add another test with your suggestion

Reply:
ok, added the test, WDYT now? Did I get it right this time or still wrong? 😅

Reply:
Yeah looks good :+1:, though the comment references `Helper` instead of `Helper2`